### PR TITLE
DB-3708: audit readmes and docs

### PR DESCRIPTION
**Pull request by CobyPear**
## What changes were made?
- update docusarus to v2.1
- update the typedoc options to have leaner API reference
- include nextjs-kit in API reference
- Fix most broken links, one lingering. The nextjs-kit modules pages recursively links to itself 🤔 not sure how to fix at the moment
- case police 🚨
- tidy up READMEs and CONTRIBUTING
- tidy up issue templates
- typedoc version was also updated, so was TypeScript
- removed circular dependency from eslint-config
- upgrade TypeScript to 4.6.x to work with the typedoc upgrade to 0.23
- fix typo in TSDoc comment in nextjs-kit

## How have the changes been tested?
READMEs everywhere have been touched up. mostly `web`
## Additional information

Do README updates need a changeset @backlineint ?

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

File: starters/gatsby-wordpress-starter/README.md [hunk 1]
```diff
@@ -48,6 +52,18 @@ Other commands:
   `lando start`).
 - `lando gatsby <command>` - run Gatsby cli commands.
 
+## Pantheon @pantheon-systems/wordpress-kit
+
+The Pantheon @pantheon-systems/wordpress-kit is included as a dependency in this
```
Comment (joa-pan) [inline starters/gatsby-wordpress-starter/README.md:57]:
```suggestion
The Pantheon `@pantheon-systems/wordpress-kit` is included as a dependency in this
```

File: starters/gatsby-wordpress-starter/README.md [hunk 2]
```diff
@@ -48,6 +52,18 @@ Other commands:
...
+project. This allows developers to make use of utility functions to simplify the
+process of building and maintaining a Front-End site on Pantheon.
+
+The `tailwindcssPlugin` is included in this project and is used to map WordPress
+Block Editor styles to Tailwind styles.
+
+Full documentation can be found at:
```
Comment (joa-pan) [inline starters/gatsby-wordpress-starter/README.md:64]:
```suggestion
For more information, see
```

File: web/docs/Packages/drupal-kit/index.md [hunk 3]
```diff
@@ -24,49 +22,36 @@ To install this package to use in your application:
 Modules can be imported from the ` @pantheon-systems/drupal-kit` package. For
 example, to use Drupal State to source data from your CMS backend:
```
Comment (joa-pan) [inline web/docs/Packages/drupal-kit/index.md:23]:
```suggestion
example, to use DrupalState to source data from your CMS backend:
```

File: web/docs/Packages/nextjs-kit/index.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+id: "index"
+title: "@pantheon-systems/nextjs-kit"
+sidebar_label: "Readme"
+sidebar_position: 0
+custom_edit_url: null
+---
+
+# Pantheon Systems Next.js Kit
+
+The `@pantheon-systems/nextjs-kit` is a collection of utilities and Next.js
```
Comment (joa-pan) [inline web/docs/Packages/nextjs-kit/index.md:11]:
```suggestion
The `@pantheon-systems/nextjs-kit` consists of utilities and Next.js
```

File: web/docs/Packages/nextjs-kit/index.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+components for use in a Pantheon Decoupled Site built with Next.js.
```
Comment (joa-pan) [inline web/docs/Packages/nextjs-kit/index.md:12]: Are we calling it Decoupled or Front-End site in this repo?

```suggestion
components for use in a Pantheon Front-End Site built with Next.js.
```

File: web/docs/Packages/nextjs-kit/index.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+## Installation
+
+To install this package to use in your application:
```
Comment (joa-pan) [inline web/docs/Packages/nextjs-kit/index.md:16]:
```suggestion
To install this package for use in your application:
```

File: README.md [hunk 7]
```diff
@@ -8,12 +8,12 @@
 - [Reporting](#bug-reports-and-feature-requests)
 - [Further Reading](#further-reading)
 
-## Prerequisites
+## Prerequisites - START HERE!
```
Comment (joa-pan) [inline README.md:11]: I don't think the "START HERE!" is needed

File: README.md [hunk 8]
```diff
@@ -23,26 +23,35 @@ locally:
   npm install -g pnpm@7.4.0
   ```
 
-## Getting Started
+---
+
+## Setup the Monorepo
```
Comment (joa-pan) [inline README.md:28]:
```suggestion
## Configure the Monorepo
```

File: README.md [hunk 9]
```diff
@@ -51,37 +60,35 @@ There are several scripts you can use on your monorepo.
...
   ```bash
   pnpm --filter './starters/**' lint
   ```
 
-- Run documentation:
-
-  ```bash
-  pnpm --filter './web' start
-  ```
+If you need to run a command in a specific project, use the alias in the root
```
Comment (joa-pan) [inline README.md:88]:
```suggestion
If you need to run a command in a specific project, use the alias in the ROOT
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -5,13 +5,13 @@ Thank you for your interest in contributing to Pantheon's Decoupled Kit ⚡️!
 ## Getting started and Prerequisites
 
 Check the
-[prerequisites](https://github.com/pantheon-systems/decoupled-kit-js/#prerequisites)
+[prerequisites](https://github.com/pantheon-systems/decoupled-kit-js/#prerequisites-start-here!)
 and
-[getting started](https://github.com/pantheon-systems/decoupled-kit-js/#getting-started)
+[getting started](https://github.com/pantheon-systems/decoupled-kit-js/#setup-the-monorepo)
 to get setup for development.
```
Comment (joa-pan) [inline CONTRIBUTING.md:11]:
```suggestion
to get set up for development.
```

**Review by backlineint — CHANGES_REQUESTED** [review]
A few minor things that I think would be worth updating:

* Could we add `sidebar_position: 1` to the root intro.md so the overview page is listed first in the menu?
* The Next.js + Drupal section seems to be missing an introduction entry. I'd imagine this could be very similar to the WP one.
* Could you add sidebar_position to the front matter for entries in the Next.js + Drupal section so the entries display in the following order: Intro, Envars, Preview, Customizations.

And could you also apply Joan's changes?

File: web/docs/Packages/nextjs-kit/index.md (hunk 5 above)
Comment (CobyPear) [reply web/docs/Packages/nextjs-kit/index.md:12]: We have a ticket to go through and clean up the names. I think we're still on the fence about where to keep Decoupled and where to switch it. I think it makes sense here, will commit this one.

**Comment by CobyPear** [comment]
I was able to get rid of the empty API reference docs from typedoc. Still some optimizations to be made there I think, but will focus on that when we move to publish these.

File: .vscode/extensions.json [hunk 11]
```diff
@@ -1,3 +1,8 @@
 {
-  "recommendations": ["esbenp.prettier-vscode", "dbaeumer.vscode-eslint"]
+  "recommendations": [
+    "esbenp.prettier-vscode",
+    "dbaeumer.vscode-eslint",
+    "streetsidesoftware.code-spell-checker",
+    "redhat.vscode-yaml"
```
Comment (CobyPear) [inline .vscode/extensions.json:6]: @backlineint any opposition to recommending these plugins? I find them very helpful, mostly the spellchecker.

**Review by backlineint — APPROVED** [review]

**Pull request merged**